### PR TITLE
Feature/apl 540 node: Peers class refactored

### PR DESCRIPTION
**Pull request by oleksiylukin**
Peers class refactored; Singletons removed from all API and peer request handlers, Welld 3.1.0

### CODE REVIEW

File: nb-configuration.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yuriy-larin-firstbridge) [inline nb-configuration.xml]: Could you pls remove it from git?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetAliases.java [hunk 2]
```diff
@@ -35,15 +35,7 @@
...
-    private static class GetAliasesHolder {
-        private static final GetAliases INSTANCE = new GetAliases();
-    }
-
-    public static GetAliases getInstance() {
-        return GetAliasesHolder.INSTANCE;
-    }
-
-    private GetAliases() {
+    public GetAliases() {
```
[the remark this answers is not in the document]
Comment (oleksiylukin) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/http/get/GetAliases.java]: done

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/ApolloCertificate.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    public String fromList(List<String> sl) {
+        String res = "";
+        for (int i = 0; i < sl.size(); i++) {
+            String semicolon = i < sl.size() - 1 ? ";" : "";
+            res += sl.get(i) + semicolon;
+        }
+        return res;
+    }
+
+    public List<String> fromString(String l) {
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/ApolloCertificate.java]: 'l' string name is not clear

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/ApolloCertificate.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            }
+        }
+        return res;
+    }
+
+    @Override
+    public String toString() {
+        String res = "Apollo X.509 Certificate:\n";
+        res += "CN=" + va.getCn() + "\n"
+                + "ApolloID=" + getApolloId().toString(16) + "\n";
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/ApolloCertificate.java]: 16 better move to constants

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/AuthorityID.java [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+      res.setRegionCode(r.getIntegerValue());
+      res.setOperationCode(o.getIntegerValue());
+      res.setSuplementalCode(s.getIntegerValue());
+      return res;
+    }
+    
+    private byte[] authorityID;
+    
+    private void setup(){
+        authorityID = new byte[16];
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/AuthorityID.java]: Move 16 to constants

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertBase.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        try {
+            String test = "Lazy Fox jumps ofver snoopy dog";
+            FBCryptoAsym ac = new AsymJCEImpl(FBCryptoParams.createDefault());
+            ac.setAsymmetricKeys(pubKey, pvtk, pubKey);
+            byte[] enc = ac.encryptAsymmetric(test.getBytes());
+            byte[] dec = ac.decryptAsymmetric(enc);
+            String test_res = new String(dec);           
+            res = test.compareTo(test_res)==0;
+        } catch (InvalidKeyException | CryptoNotValidException ex) {
+        }
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertBase.java]: Maybe print some message in trace or debug mode?

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                if (f.isFile() && f.canRead()) {
+                    ApolloCertificate vc = cl.loadCertificate(f.getAbsolutePath());
+                    if (vc != null) {
+                        put(vc.getApolloId(), vc);
+                        if (loadPrivateKey) {
+                            FilePath fp = new FilePath(f);
+                            String parent = fp.getPath();
+                            String fn = fp.getFileName();
+                            PrivateKey pk = null;
+                            pk = readPvtKey(parent + "/" + pvtKeyFileName(fn));
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java]: Is '/' file delimiter or resource delimiter?

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    public static String rmSuffixes(String fn) {
+        String name = new String(fn);
+        FilePath fp = new FilePath(fn);
+        String ext = fp.getExtension();
+        if (ext.equalsIgnoreCase("pem")) {
+            name = fp.getName();
+        } else {
+            name = fn;
+        }
+        String[] sfxes = {"_pvtkey", "_req", "_cert", "_selfcert", "_csr"};
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java]: sfxes should be moved to constants

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    public static String certFileName(String fn) {
+        String suffix = "_cert";
+        String name = rmSuffixes(fn);
+        return name + suffix + ".pem";
+    }
+
+    public static String csrFileName(String fn) {
+        String suffix = "_csr";
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/CertificateHolder.java]: Move string literals  to constants

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/TrustAllSSLProvider.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import java.security.cert.X509Certificate;
+
+public class TrustAllSSLProvider {
+
+    // Verify-all name verifier
+    private final static HostnameVerifier hostNameVerifier = (hostname, session) -> true;
+
+    // Trust-all socket factory
+    private static final SSLSocketFactory sslSocketFactory;
+    static {
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cert/TrustAllSSLProvider.java]: Maybe avoid using static block here?

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cls/ClsItem.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        return lv;
+    }
+
+    public BigInteger getBigIntegerVAlue() {
+        BigInteger res = BigInteger.ZERO;
+        try {
+            res = new BigInteger(value);
+        } catch (NumberFormatException e10) {
+            try {
+                res = new BigInteger(value, 16);
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/cls/ClsItem.java]: 16 and 8 move to constants

File: apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/env/FilePath.java [hunk 12]
```diff
@@ -0,0 +1,70 @@
+package com.apollocurrency.aplwallet.apl.util.env;
+
+import java.io.File;
+
+/**
+ *
+ * @author alukin@gmail.com
+ */
+public class FilePath {
```
Comment (AndrewBoyarsky) [inline apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/env/FilePath.java]: Do we require this class? Can we use Path instead?

**Pull request merged**